code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -1023,3 +1022,16 @@
 
 /mob/living/silicon/ai/zMove(dir, feedback = FALSE)
 	. = eyeobj.zMove(dir, feedback)
+
+
+/// Proc to hook behavior to the changes of the value of [aiRestorePowerRoutine].
+/mob/living/silicon/ai/proc/set_aiRestorePowerRoutine(new_value)
+	if(new_value == aiRestorePowerRoutine)
+		return
+	. = aiRestorePowerRoutine
+	aiRestorePowerRoutine = new_value
+	if(aiRestorePowerRoutine)
+		if(!.)
+			ADD_TRAIT(src, TRAIT_INCAPACITATED, POWER_LACK_TRAIT)
+	else if(.)
```

Comment:
Kinda the same thing

Reply:
You'll see this pattern on a lot of procs. It basically saves as a return the old state, makes the change, and then compares both, allowing things calling this to know about changes if any.